File: 0226.InvertBinaryTree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def invertTree(self, root):
        """
        :type root: TreeNode
        :rtype: TreeNode
        """
        # recursive
        if not root:
            return None
        '''
        l = self.invertTree(root.left)
        r = self.invertTree(root.right)
        root.left = r
        root.right = l
        '''
        root.left, root.right = self.invertTree(root.right), self.invertTree(root.left)
        return root
        """
        # BFS/iteratively with queue
        queue = [root]
        while queue:
            node = queue.pop(0)
            if node:
                node.left, node.right = node.right, node.left
                queue += [node.left, node.right]
        return root
        """

File: 0000-0999/0226.InvertBinaryTree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func invertTree(root *TreeNode) *TreeNode {
    if root == nil {
        return root
    }
    root.Left, root.Right = invertTree(root.Right), invertTree(root.Left)
    /* queue
    queue := []*TreeNode{root}
    for i := 0 ; i < len(queue); i++ {
        node := queue[i]
        if node != nil {
            node.Left, node.Right = node.Right, node.Left
            queue = append(queue, node.Left, node.Right)
        }
    }*/
    return root
}